var/spack/repos/builtin/packages/zfp/package.py

```diff
@@ -103,28 +103,32 @@ class Zfp(CMakePackage, CudaPackage):
     )
 
     variant(
-        "tight_error",
+        "tight-error",
         default=False,
-        description="EXPERIMENTAL: Enable rounding-based error reduction",
+        description="EXPERIMENTAL: Use tighter error bound when rounding is enabled",
     )
 
     variant("twoway", default=False, description="Use two-way skew-associative cache")
 
     # Conflicts
     conflicts("+c", when="@:0.5.3", msg="+c requires zfp 0.5.4 or later")
-    conflicts("+daz", when="@:0.5.5", msg="+daz required zfp 1.0.0 or later")
-    conflicts("+python", when="@:0.5.4", msg="+c requires zfp 0.5.4 or later")
-    conflicts("+fortran", when="@:0.5.4", msg="+c requires zfp 0.5.4 or later")
-    conflicts("+openmp", when="@:0.5.2", msg="+c requires zfp 0.5.4 or later")
-    conflicts("+cuda", when="@:0.5.3", msg="+c requires zfp 0.5.4 or later")
+    conflicts("+daz", when="@:0.5.5", msg="+daz requires zfp 1.0.0 or later")
+    conflicts("+python", when="@:0.5.4", msg="+python requires zfp 0.5.4 or later")
+    conflicts("+fortran", when="@:0.5.4", msg="+fortran requires zfp 0.5.4 or later")
+    conflicts("+openmp", when="@:0.5.2", msg="+openmp requires zfp 0.5.4 or later")
+    conflicts("+cuda", when="@:0.5.3", msg="+cuda requires zfp 0.5.4 or later")
     conflicts("+fasthash", when="@:0.5.1", msg="+fasthash requires zfp 0.5.2 or later")
     conflicts("+profile", when="@:0.5.1", msg="+profile requires zfp 0.5.2 or later")
-    conflicts("round", when="@:0.5.5", msg="+round requires zfp 1.0.0 or later")
-    conflicts("+tight_error", when="@:0.5.5", msg="+tight_error requires zfp 1.0.0 or later")
+
+    conflicts("round=never", when="@:0.5.5", msg="round requires zfp 1.0.0 or later")
```

Comment:
This is the default behavior for zfp 1.0.0 and all prior versions and should probably be accepted unconditionally.

@tldahlgren Does Spack have a policy in situations like this?

Reply:
If this value is acceptable for versions supporting the variant, then you don't want the package to have a conflict.

Reply:
Just to be clear, earlier versions of the zfp spack recipe did not support this variant.  The behavior prior to zfp 1.0.0 is identical to the newly introduced variant `round=never`.  zfp 1.0.0 adds new features that correspond to `round=first` and `round=last`, with `round=never` remaining the default.

The question I had is if it's considered poor practice to now allow `round=never` with older versions, which prior to zfp 1.0.0 was really the one and only rounding option, but also something users did not even know about.  I don't feel strongly one way or the other, but I suspect this issue will recur for us in the future and would like to know if there's a recommended approach to handling conflicts for new variants with respect to old packages.

Reply:
@lindstro: I'm not aware of a recommend approach in this regard. Do you approve this PR now?

Reply:
Isn't there the ability to add a when clause to a variant?  It seems like the answer here is to make the round variant when="@1.0.0:" and not bother with a conflicts clause

Reply:
Yes, we have done so with some variants but not consistently with all of them.  I agree with your suggestion.